test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/EmailControllerSpec.scala

```diff
@@ -61,40 +60,32 @@ class EmailControllerSpec extends ControllerSpec with AuthSupport with SessionSu
 
   implicit lazy val messagesApi: MessagesApi = controller.messagesApi
 
-  val nino                 = NINO("AB123456C")
-  val name                 = models.Name("forename", "surname")
-  val retrievedDateOfBirth = new JodaLocalDate(2000, 4, 10)
+  val nino = NINO("AB123456C")
+
+  val name = Name("forename", "surname")
+
+  val dateOfBirth = new JodaLocalDate(2000, 4, 10)
+
+  val bpr = sample[BusinessPartnerRecord]
 
   val subscriptionDetails = sample[SubscriptionDetails]
 
   def mockUuidGenerator(uuid: UUID): CallHandler0[UUID] =
     (mockUuidGenerator.nextId: () => UUID).expects().returning(uuid)
 
   def mockEmailVerification(expectedEmail: Email, expectedId: UUID, expectedName: String)(
-    result: Either[Error, EmailVerificationResponse]
-  ) =
+    result: Either[Error, EmailVerificationResponse]) =
     (mockService
       .verifyEmail(_: Email, _: UUID, _: String)(_: HeaderCarrier))
       .expects(expectedEmail, expectedId, expectedName, *)
       .returning(EitherT.fromEither[Future](result))
 
-  val retrievals = Retrievals.nino and Retrievals.itmpName and Retrievals.name and Retrievals.itmpDateOfBirth
-  val retrievalsResult = Future successful (
-    new ~(
-      new ~(
-        new ~(Some("nino"), Some(ItmpName(Some("givenName"), Some("middleName"), Some("familyName")))),
-        Some(Name(Some("forename"), Some("surname")))
-      ),
-      Some(new JodaLocalDate(2000, 4, 10))
-    )
-  )
-
-  def subscriptionDetailsBehavior(performAction: () => Future[Result]): Unit =
+  def subscriptionDetailsBehavior(performAction: () => Future[Result]): Unit = {
     "redirect to check your details" when {
 
       "there is no session data" in {
         inSequence {
-          mockAuthWithCl200AndRetrievedAllRetrievals(nino.value, name, retrievedDateOfBirth)
+          mockAuthWithCl200AndRetrievedAllRetrievals(nino.value, name, dateOfBirth)
           mockGetSession(Future.successful(Right(None)))
```

Comment:
let's rename `mockAuthWithCl200AndRetrievedAllRetrievals` to `mockAuthWithCl200AndWithAllRetrievals` - I couldn't think of a better name at the time.

Reply:
that is a better name - will refactor this

Reply:
LGTM